reactor-core/src/main/java/reactor/core/publisher/Mono.java

```diff
@@ -2116,7 +2150,7 @@ public T block(Duration timeout) {
 	 * otherwise
 	 */
 	public final Mono<Boolean> hasElement() {
-		return onAssembly(new MonoHasElements<>(this));
+		return onAssembly(new MonoHasElement<>(this));
```

Comment:
this MonoHasElement change is orthogonal... shelve it and isolate it in another PR / commit on master